overlord/snapstate/handlers.go

```diff
@@ -3500,14 +3500,60 @@ func (m *SnapManager) doSetAutoAliases(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	if !snapst.AliasesPending {
+		// remove-aliases didn't run.
+		// set this flag to let setup-aliases know that it should
+		// prune old aliases.
+		//
+		// NOTE: this will also trigger for first installs since
+		// AliasesPending will also be false for a new install but
+		// it is fine since prunning without without existing aliases
+		// is basically a no-op.
+		t.Set("prune-old-aliases", true)
+	}
```

Comment:
@pedronis I am not sure if this is okay, but since we depend on `AliasesPending` which will be unset during first install, this triggers the prunning flow during first install, which doesn't hurt.

just wanted to point it out, please let me know if you have concerns or other suggestions.

Reply:
that's fine but you could also change the if condition to: !snaps.AliasesPending && len(curAliases) != 0, no?

Reply:
@pedronis unfortunately `!snaps.AliasesPending && len(curAliases) != 0` gives a corner case were `remove-aliases` could have been skipped (for a snap with no aliases) and prune mode is not initiated for `setup-aliases`.

Reply:
@bboozzoo agree, this part needs to be clear, thank you!